plugin/src/main/java/org/opensearch/ml/cluster/MLSyncUpCron.java

```diff
@@ -223,7 +227,7 @@ private void undeployExpiredModels(
 
     @VisibleForTesting
     void initMLConfig() {
-        if (mlConfigInited) {
+        if (mlConfigInited || mlFeatureEnabledSetting.isMultiTenancyEnabled()) {
             return;
         }
         mlIndicesHandler.initMLConfigIndex(ActionListener.wrap(r -> {
```

Comment:
Change this part like EncryptorImpl.java?

Reply:
For multi-tenancy, init master key will happen only when the first request comes with tenant id. The cron job doesn't know tenant id, so we have to skip it directly for Multi-tenancy use case.